experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -201,6 +218,77 @@ export class EditManager<
 		});
 	}
 
+	/**
+	 * Makes the given branch known to the `EditManager` as a revertible branch. This ensures that the `EditManager` will
+	 * keep track of the oldest revertible sequence ID on this branch and will not trim any commits after it.
+	 *
+	 * @privateRemarks
+	 * TODO: this is not currently used and is included for documentation purposes
+	 * This will need to be properly implemented or replaced with a better alternative. Branches may hold onto revertible commits
+	 * that have been sequenced and could possibly be evicted from the trunk.
+	 */
+	private registerRevertibleBranch(branch: SharedTreeBranch<TEditor, TChangeset>): void {
+		// if the branch doesn't have any revertibles,
+		if (branch.hasRevertibleCommits()) {
+			const oldest = this.getOldestRevertibleSequenceId(branch);
+			const branches = getOrCreate(this.revertibleBranches, oldest, () => new Set());
```

Comment:
What happens if `oldest` is undefined?

Combined with the suggestion at the other place where we call `hasRevertibleCommits`, you could just get rid of the `hasRevertibleCommits` method entirely and rely on checking `this.getOldestRevertibleSequenceId` for defined-ness instead.